modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/ExclusiveConsumerTestCase.java

```diff
@@ -0,0 +1,171 @@
+     * 2. Create two subscribers for the "exclusiveQueue"
+     * 3. Publish 1000 messages to queue.
+     * 4. First consumer should receive all the 1000 messages.
+     * 5. Second consumer should not receive any messages
+     *
+     * @throws org.wso2.mb.integration.common.clients.exceptions.AndesClientConfigurationException
+     * @throws javax.jms.JMSException
+     * @throws javax.naming.NamingException
+     * @throws java.io.IOException
+     * @throws org.wso2.mb.integration.common.clients.exceptions.AndesClientException
+     */
+    @Test(groups = "wso2.mb", description = "Exclusive Consumer test case")
+    public void performSingleQueueSendReceiveTestCase()
+            throws AndesClientConfigurationException, JMSException, NamingException, IOException,
+                   AndesClientException, LoginAuthenticationExceptionException, URISyntaxException,
+                   LogoutAuthenticationExceptionException, XMLStreamException,
+                   AndesAdminServiceBrokerManagerAdminException, SAXException,
+                   TopicManagerAdminServiceEventAdminExceptionException, XPathExpressionException,
+                   UserAdminUserAdminException {
+        long sendCount = 1000L;
+        long expectedCount = 1000L;
+        createQueueWithExclusiveConsumer("exlusiveQueue");
+        // Creating a consumer client configuration
+        AndesJMSConsumerClientConfiguration consumerConfig = new AndesJMSConsumerClientConfiguration(ExchangeType.QUEUE, "exlusiveQueue");
+        consumerConfig.setPrintsPerMessageCount(expectedCount / 10L);
+        // Creating a publisher client configuration
+        AndesJMSPublisherClientConfiguration publisherConfig = new AndesJMSPublisherClientConfiguration(ExchangeType.QUEUE, "exlusiveQueue");
+        publisherConfig.setNumberOfMessagesToSend(sendCount);
+        publisherConfig.setPrintsPerMessageCount(sendCount / 10L);
+        // Creating first client
+        AndesClient consumerClient1 = new AndesClient(consumerConfig, true);
+        consumerClient1.startClient();
+        // Creating second client
+        AndesClient consumerClient2 = new AndesClient(consumerConfig, true);
+        consumerClient2.startClient();
+        AndesClient publisherClient = new AndesClient(publisherConfig, true);
+        publisherClient.startClient();
+        AndesClientUtils.waitForMessagesAndShutdown(consumerClient1, AndesClientConstants.DEFAULT_RUN_TIME);
+        AndesClientUtils.waitForMessagesAndShutdown(consumerClient2, AndesClientConstants.DEFAULT_RUN_TIME);
+        // Evaluating
+        Assert.assertEquals(publisherClient.getSentMessageCount(), sendCount, "Error in sending all the messages.");
+        Assert.assertEquals(consumerClient1.getReceivedMessageCount(), expectedCount, "Error in sending all the messages to the exclusive consumer.");
+        Assert.assertEquals(consumerClient2.getReceivedMessageCount(), 0, "Error in sending messages to non-exclusive consumer.");
```

Comment:
I think the error message should be more meaningful like non-exclusive subscriber received messages.